### ISSUE TITLE
Difference between visits in graph and visits in table

### ISSUE THREAD

**Issue by Mich-es** [matomo-org/plugin-GoogleAnalyticsImporter#46]
After i import some data, i see a difference between visits and visits:
![grafik](https://user-images.githubusercontent.com/58422824/70785781-81966c00-1d8b-11ea-9d04-226c1b1dbe04.png)

**Comment by diosmosis** [comment]
Can you export the visitors overview graph as XML and paste the result here?

**Comment by diosmosis** [comment]
Could you also provide screenshots for the bottom metrics for the 3 day and full month ranges so I can compare the results?

**Comment by diosmosis** [comment]
Also are you currently tracking into this site? And does this happen with older imported data? Eg, a couple months in the past.

**Comment by Mich-es** [comment]
> 
> 
> Could you also provide screenshots for the bottom metrics for the 3 day and full month ranges so I can compare the results?

![grafik](https://user-images.githubusercontent.com/58422824/70823698-d9ab8d80-1de0-11ea-999e-275e34ad2841.png)

**Comment by Mich-es** [comment]
> 
> 
> Also are you currently tracking into this site? And does this happen with older imported data? Eg, a couple months in the past.

Yes, current figures run parallel to the import in the last 3 days.

I am currently importing 4 accounts and this phenomenon occurs with all accounts. I have the feeling that it is not properly "archived" !?

**Comment by diosmosis** [comment]
Can you check if the data inaccuracy happens when you select older months? For example, on October 2019, is the data different? (if yes, an export and screenshot would be useful)

Btw, it looks like your metrics are being affected by this bug as well: https://github.com/matomo-org/matomo/issues/15200 (the fix is here: https://github.com/matomo-org/matomo/pull/15218 )

I think this is an issue we found a day or two ago that occurs w/ the latest matomo version. I'm currently working on a fix for it, but I need a bit more information to completely diagnose it.

**Comment by Mich-es** [comment]
> Can you check if the data inaccuracy happens when you select older months? For example, on October 2019, is the data different? (if yes, an export and screenshot would be useful)

no, the import of the accounts goes until 2017. But the same effect can be seen everywhere. As soon as the period is increased, the number of visitors in the table no longer matches the graphic


> Btw, it looks like your metrics are being affected by this bug as well: [matomo-org/matomo#15200](https://github.com/matomo-org/matomo/issues/15200) (the fix is here: [matomo-org/matomo#15218](https://github.com/matomo-org/matomo/pull/15218) )

I compare the files, but i can´t find the fix!?

> I think this is an issue we found a day or two ago that occurs w/ the latest matomo version. I'm currently working on a fix for it, but I need a bit more information to completely diagnose it, in case it's actually something else.

I would like help you - tell me how :-)

**Comment by Mich-es** [comment]
Another issue, same problem?

![grafik](https://user-images.githubusercontent.com/58422824/70828925-b25abd80-1dec-11ea-8875-22246280d48a.png)
![grafik](https://user-images.githubusercontent.com/58422824/70828959-c7375100-1dec-11ea-9fe4-ac8f975fcebf.png)

**Comment by diosmosis** [comment]
> I compare the files, but i can´t find the fix!?

It's the code that's removed on the left. It format's the metrics twice which results in some of the ending digits getting cut off. The fix is to remove the redundant code. You can get a patch file via https://patch-diff.githubusercontent.com/raw/matomo-org/matomo/pull/15218.patch .

> I would like help you - tell me how :-)

If it applies to old data too then it's not the issue the team found elsewhere. Can you apply the above patch (by removing the code) and see if it changes things for you?

> Another issue, same problem?

You're looking at different metrics, in GA "users" is equivalent to Matomo "visitors" (which is what is pictured at the top if google translate worked), but in the bottom graph you have "visits", not "visitors".

**Comment by diosmosis** [comment]
To apply the change you could also simply replace the `plugins/VisitsSummary/Reports/Get.php` file with  https://raw.githubusercontent.com/matomo-org/matomo/3.x-dev/plugins/VisitsSummary/Reports/Get.php .

**Comment by Mich-es** [comment]
> It's the code that's removed on the left. It format's the metrics twice which results in some of the ending digits getting cut off. The fix is to remove the redundant code. You can get a patch file via https://patch-diff.githubusercontent.com/raw/matomo-org/matomo/pull/15218.patch .

If the effect is to occur immediately, the patch will not work. I ran the cronjob, but there was no effect either. Do I have to reset something, or does that also affect the archived data?


> You're looking at different metrics, in GA "users" is equivalent to Matomo "visitors" (which is what is pictured at the top if google translate worked), but in the bottom graph you have "visits", not "visitors".

Oh, that's actually a misunderstanding - maybe also in the translation.
In Google, these are sessions, not (unique) visitors. I changed that at sessions at Goole and now the numbers are right

**Comment by diosmosis** [comment]
> Do I have to reset something, or does that also affect the archived data?

It just affects how the data is displayed in the metrics section under the graph. For example, in the screenshot you posted there are 26,86 visits, but you can't have 0,86 visits. The bug is causing the 

You can try clearing the cache via `./console cache:clear` or restarting the web server to make sure the changed PHP file is used, but it should be immediately visible.

**Comment by Mich-es** [comment]
I restart the server - no effect
i clear the chache - no effect

**Comment by Mich-es** [comment]
Have a look at the periods:
![grafik](https://user-images.githubusercontent.com/58422824/70845948-cb975480-1e54-11ea-9aa7-aa600bd1416e.png)

![grafik](https://user-images.githubusercontent.com/58422824/70845962-e4076f00-1e54-11ea-9bd7-4815df3080d7.png)

![grafik](https://user-images.githubusercontent.com/58422824/70845969-f41f4e80-1e54-11ea-8387-38d16763822d.png)

**Comment by diosmosis** [comment]
Can you also show the date time selected in the period selector for those graphs?

**Comment by Mich-es** [comment]
Do you mean this?
![grafik](https://user-images.githubusercontent.com/58422824/70846477-eb7d4700-1e59-11ea-9136-53655ccfa4b2.png)

**Comment by Mich-es** [comment]
Oh, i play a little bit see what happens:
![grafik](https://user-images.githubusercontent.com/58422824/70846517-4dd64780-1e5a-11ea-8ce4-7a171026ae40.png)

![grafik](https://user-images.githubusercontent.com/58422824/70846558-aad1fd80-1e5a-11ea-81a6-2292262f03ae.png)

**Comment by diosmosis** [comment]
In your first image you have selected a range, the metrics at the bottom are the summed metrics for each day in the range.

For the second and third, the metrics are for the selected period, the graph shows periods in the past. They are 0 because there is 0 visits for the selected period (ie, 0 visits for March 2012 and 0 visits for the year 2012).

I think your problems with metrics are fixed now?

**Comment by Mich-es** [comment]
> 
> 
> In your first image you have selected a range, the metrics at the bottom are the summed metrics for each day in the range.
> 
> For the second and third, the metrics are for the selected period, the graph shows periods in the past. They are 0 because there is 0 visits for the selected period (ie, 0 visits for March 2012 and 0 visits for the year 2012).

The periods of the graphics and the summation date from **2012.** The detailed dates from **March 2012**. To compare this, I added a screenshot of GA.
![grafik](https://user-images.githubusercontent.com/58422824/70847873-e0caae00-1e69-11ea-94e9-3607fe78323f.png)


> I think your problems with metrics are fixed now?
Yes, meanwhile i think so :-)

**Comment by diosmosis** [comment]
If you're referring to data being 0 for 2012 when it's not in GA, can you check the import status for the last day imported?

**Comment by Mich-es** [comment]
> can you check the import status for the last day imported?
![grafik](https://user-images.githubusercontent.com/58422824/70860547-55155800-1f23-11ea-8d30-5df534f78d96.png)


Unfortunately I didn't understand the last question. What I wanted to show is that if the period is 3 days, the dates are fine. If I extend the period, the number from the 3-day period remains. If I extend the period to one month, everything is set to 0. I think it's a bug, but I can't say whether it's caused by Google Analytics Importer or Matomo itself.

These are enormous amounts of data, so the import will certainly take 12 months. The import data are currently available until August 2012.

In parallel, the tracking for the pages has been running for 4 days.

**Comment by diosmosis** [comment]
So there is data for 2012 and everything is working as expected now?

**Comment by Mich-es** [comment]
> So there is data for 2012 and everything is working as expected now?

No, not at all, as you can see in the graphic above.

18 visitors are shown in the overview, while the graph shows over 2.8 million.

It doesn't work as expected

**Comment by diosmosis** [comment]
@Mich-es if you're using a "range" period it adds up the visits for every day. If you're just on a specific day it shows the value for the last day in the graph. Please check the tooltip for the very last point on the right side of the graph.

**Comment by Mich-es** [comment]
> if you're using a "range" period it adds up the visits for every day. If you're just on a specific day it shows the value for the last day in the graph. Please check the tooltip for the very last point on the right side of the graph.

and that’s exactly what doesn’t work. did you look at the graphic above?

I will show this again here.

A time series with 2.8 million users (January 1, 2012 - August 31, 2012).

The summary below the graphic shows 0 instead of 2.8 million - the function is incorrect or does not work correctly.

Or do I just not understand you? :-)

![grafik](https://user-images.githubusercontent.com/58422824/70861652-61081680-1f31-11ea-951c-0f40ae0ccf9c.png)

**Comment by diosmosis** [comment]
Your server doesn't have enough memory to aggregate a range that large, so it fails and reports 0. Given this is not a problem with the importer I'm going to close the issue and refer you to the forums: http://forums.matomo.org/.
